rust/src/storage/file/mod.rs

```diff
@@ -87,6 +87,12 @@ impl StorageBackend for FileStorageBackend {
     async fn put_obj(&self, path: &str, obj_bytes: &[u8]) -> Result<(), StorageError> {
         let tmp_path = create_tmp_file_with_retry(self, path, obj_bytes).await?;
 
+        let dir = std::path::Path::new(path);
+
+        if let Some(d) = dir.parent() {
+            fs::create_dir_all(d).await?;
+        }
+
```

Comment:
Yeap, I've notice that too, but my recent PR changes the way put_obj and rename_obj should work, please see https://github.com/delta-io/delta-rs/pull/197/files#diff-ce6ba0c9d799f33c7c8ec4b5d7f36e1c03c8451390190a348b530caf838c89dfR87-R88

Because current version of put_obj of file brings ambiguity, e.g. it's not actual put_obj but the whole ceremony

Reply:
But I'm happy to merge your changes and I'll update my PR then